### PR TITLE
#2554 configure user session timeout

### PR DESCRIPTION
**Pull request by darddan**
**Description**
Session timeout is hardcoded to 2 hours. This PR makes that value configurable with a fallback to 2 hours. Related to #2554

There are two use-cases for me for this change. The first is that I wan't to be able to stay logged in for longer. The other one is if I'm downloading really large files and session expires in between then the upload is canceled at that point. 

IMO for the second use-case this PR shouldn't be enough and it needs another proper fix.

### CODE REVIEW

**Comment by darddan** [comment]
Hi @o1egl, can I get a review/comment on the changes? I fixed the linting issues after the CI failed last time

File: http/auth.go [hunk 1]
```diff
@@ -172,11 +174,13 @@ var signupHandler = func(w http.ResponseWriter, r *http.Request, d *data) (int,
...
 }
 
-var renewHandler = withUser(func(w http.ResponseWriter, r *http.Request, d *data) (int, error) {
-	return printToken(w, r, d, d.user)
-})
+func renewHandler(tokenExpireTime time.Duration) handleFunc {
+	return withUser(func(w http.ResponseWriter, r *http.Request, d *data) (int, error) {
+		return printToken(w, r, d, d.user, tokenExpireTime)
+	})
+}
```
Comment (o1egl) [inline http/auth.go]: Please adjust it with the fix in master https://github.com/filebrowser/filebrowser/blob/master/http/auth.go#L175-L178

**Comment by StanislavJochman** [comment]
Where can I set a custom user session timeout? Is it only through API?

**Comment by akamkomputer** [comment]
Is it possible to set TokenExpirationTime in docker?

**Comment by fritx** [comment]
Update: I've submitted a PR for this feature. https://github.com/filebrowser/filebrowser/pull/2845

---

cfg file should work @StanislavJochman @akamkomputer 

```sh
filebrowser -c your/path/to/filebrowser.json
```

```js
// filebrowser.json
{
  "port": 8080,
  "baseURL": "",
  "address": "127.0.0.1",
  "log": "stdout",
  "database": "filebrowser.db",
  "root": ".",
  "tokenExpirationTime": "12h"
}
```

**Comment by StanislavJochman** [comment]
But can it be set using dockerfile? Or do I have to edit file manually?

**Comment by fritx** [comment]
@StanislavJochman oops, my fault/.. Update: I've submitted a PR for this feature. https://github.com/filebrowser/filebrowser/pull/2845

```sh
# One of the below should work:
filebrowser --token-expiration-time=12h
# or
filebrowser -c fb.json
# fb.json
{
  "token-expiration-time": "12h"
}
```

UPDATE:  corrected: `filebrowser --token-expiration-time=12h`, as pointed out by baur.

**Comment by baur** [comment]
implemented: 
```
filebrowser --token-expiration-time=12h
```
not **timeout**

**Comment by k0gen** [comment]
> implemented:
> 
> ```
> filebrowser --token-expiration-time=12h
> ```
> 
> not **timeout**

What are the allowable values and range here? Can "0" be used to disable the timeout?

**Comment by MattDHill** [comment]
@o1egl Can you confim how we can disable the expiration altogether such that sessions never expire?

**Pull request merged**